zap/src/main/java/org/zaproxy/zap/extension/script/ExtensionScript.java

```diff
@@ -1039,6 +1039,9 @@ private int addScriptsFromDir(File dir, ScriptType type, String targetEngineName
                                             type,
                                             false,
                                             f);
+                            if (this.getScriptParam().isEnableScriptsFromDirs()) {
+                                this.setEnabled(sw, true);
+                            }
```

Comment:
As mentioned in the issue, (IMO) this should only apply to directories specified through the options not to all the directories added.

Reply:
I think i dont understand this right. What other directories will this be called on (besides the one specified through the options)?

Reply:
Oh, the default one i suppose?

Reply:
The add-ons can add their own directories (e.g. Community Scripts).

Reply:
Fair enough. I'm guessing just an `and` check in that if for whether the current dir is in `getScriptParam().getScriptDirs()` should suffice?

Reply:
Cleaner if the enabled state is specified when adding the options directories (see `postInit`), rather than do extra checks here.

Reply:
If I understand correctly, that would mean modifying `addScriptsFromDir()` to also accept an optional bool argument?

Reply:
Correct, note that it needs to keep binary compatibility (add a new method).

Reply:
Take a look @ https://github.com/zaproxy/zaproxy/pull/7623/commits/eb1ad4dc8e2c0fed234b5f9ade83966c74872690